### ISSUE TITLE
Clarification: is browser autocomplete *alone* sufficient as a mechanism for accessible authentication?

### ISSUE THREAD

**Issue by brothercake** [w3c/wcag#3550]
This relates to a recent case I tested (the Co-op online banking interface, fwiw) in which copy-and-paste is blocked by scripting, but browser autocomplete still works. I considered this a pass, but I'm not entirely sure that was right.

So is autocomplete alone sufficient for "a mechanism is available to assist the user in completing the cognitive function test" or would the lack of copy-paste support constitute a failure nonetheless?

Either way, can (should) this be clarified in understanding docs?

**Comment by patrickhlauke** [comment]
my personal take here is no, that's not enough, because a user may well be using a separate 3rd party password manager that requires them to copy from the app, then manually paste into the field on the website in the browser (though yes, the SC doesn't necessarily say which method/solution must be supported)

**Comment by mbgower** [comment]
Looking at the current wording for Accessible Authentication and its techniques, I think your scenario is at least not a clear failure, and seems to me to be sufficient, based on wording:
1. WCAG 1[.3.5 Identify Input Purpose](https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html#input-purposes) specifically calls out username and current password as in scope. So if those are implemented, the user who has previously entered the strings will theoretically be able to choose those again.
2. Step 1 of the failure technique for [F109 Failure of Success Criterion 3.3.8 and 3.3.9 due to preventing password or code re-entry in the same format](https://www.w3.org/WAI/WCAG22/Techniques/failures/F109) specifically mentions auto-filling
> Check if the structure of the input field(s) prevents the user from pasting or auto-filling the entire password or code in the format in which it was originally created.

So it comes down to the use/interpretation of "or" there, as well as the second bullet (not quoted). Given autocomplete appears to meet the Mechanism bullet, quoted below, it seems like it should be sufficient (and that "or" should probably be an "and")
> A [mechanism](https://www.w3.org/WAI/WCAG22/Understanding/accessible-authentication-minimum#dfn-mechanism) is available to assist the user in completing the cognitive function test.

It's nothing I'd champion -- I completely agree being unable to paste in a paste word is a _real_ barrier --  but the scenario definitely bears discussion.
